conans/test/generators/pkg_config_test.py

```diff
@@ -55,3 +58,94 @@ def variables_setup_test(self):
 Libs: -L${libdir}
 Cflags: -I${includedir} -Flag1=23 -DMYDEFINE1
 """)
+
+    def pkg_config_dirs_test(self):
+        # https://github.com/conan-io/conan/issues/2756
+        conanfile = """
+import os
+from conans import ConanFile
+
+class PkgConfigConan(ConanFile):
+    name = "MyLib"
+    version = "0.1"
+
+    def package_info(self):
+        self.cpp_info.includedirs = []
+        self.cpp_info.libdirs = []
+        self.cpp_info.bindirs = []
+        self.cpp_info.libs = []
+        libname = "mylib"
+        fake_dir = os.path.abspath(os.path.join("my_absoulte_path", "fake"))
+        include_dir = os.path.join(fake_dir, libname, "include")
+        lib_dir = os.path.join(fake_dir, libname, "lib")
+        self.cpp_info.includedirs.append(include_dir)
+        self.cpp_info.libdirs.append(lib_dir)
+"""
+        client = TestClient()
+        client.save({"conanfile.py": conanfile})
+        client.run("create . danimtb/testing")
+        client.run("install MyLib/0.1@danimtb/testing -g pkg_config")
+
+        pc_path = os.path.join(client.current_folder, "MyLib.pc")
+        self.assertTrue(os.path.exists(pc_path))
+        pc_content = load(pc_path)
+        self.assertNotIn("${prefix}", pc_content)
+
+        def assert_is_abs(path):
+            self.assertTrue(os.path.isabs(path))
+
+        for line in pc_content.splitlines():
+            if line.startswith("includedir="):
+                assert_is_abs(line[len("includedir="):])
+                self.assertTrue(line.endswith("include"))
+            elif line.startswith("libdir="):
+                assert_is_abs(line[len("libdir="):])
+                self.assertTrue(line.endswith("lib"))
+
+    def pkg_config_rpaths_test(self):
+        # rpath flags are only generated for gcc and clang
+        profile = """
+[settings]
+os=Linux
+os_build=Linux
+arch=x86_64
+arch_build=x86_64
+compiler=gcc
+compiler.version=7
+compiler.libcxx=libstdc++
+build_type=Release
+[options]
+[build_requires]
+[env]
+"""
+        conanfile = """
+import os
+from conans import ConanFile
+
+class PkgConfigConan(ConanFile):
+    name = "MyLib"
+    version = "0.1"
+    settings = "os", "compiler", "arch", "build_type"
```

Comment:
Why don't you remove the settings and the profile? the generator seems to don't care about any setting. This way the package id is the same too.

Reply:
rpaths are only used with gcc and clang:
https://github.com/danimtb/conan/blob/37144b9a4718ce6e8a22aa7bbf35df82cbe19052/conans/client/build/compiler_flags.py#L16#23